test/optim/test_optim.py

```diff
@@ -48,6 +54,75 @@ def _diff_fn(p, grad, opt_differentiable_state, opt_class, kwargs, *ignored):
     )
 
 
+def _multistep_backprop_diff_lr_fn(
+    params: Tensor,
+    grad: Tensor,
+    inner_opt_differentiable_state: dict[str, Any],
+    inner_opt_class: type[Optimizer],
+    inner_kwargs: dict[str, Any],
+    *ignored: Any,
+) -> tuple[Tensor, ...]:
+    assert (
```

Comment:
Is there a decision you made here where the caller function is responsible for handling differentiable = True? i.e., another way to do this is to

```
inner_kwargs["differentiable"] = True
```

instead of the assert and have the caller function not pass in differentiable at all.

Reply:
I was thinking that every time the prexisting _diff_fn is called, differentiable is True, so I wanted to follow the convention and have it be explicitly passed in inner_kwargs. At the same time, I knew the test would break if it wasn't true so I thought I should add the assert

Reply:
okay, i'm fine with that